From 8ec9f75afc8327b5a61b1c75837a5cda55c72318 Mon Sep 17 00:00:00 2001
From: Les Hill and Sandro Turriate <dev+leshill+sandro@hashrocket.com>
Date: Mon, 11 Oct 2010 13:28:27 -0400
Subject: [PATCH] Respect consider_all_requests_local setting

- this is explicitly set in all the generated environment files
- ShowExceptions uses false as a default
- also resolves issue #5166
---
 .../action_dispatch/middleware/show_exceptions.rb  |    2 +-
 actionpack/test/abstract_unit.rb                   |    2 +-
 actionpack/test/dispatch/show_exceptions_test.rb   |    8 ++++----
 3 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/actionpack/lib/action_dispatch/middleware/show_exceptions.rb b/actionpack/lib/action_dispatch/middleware/show_exceptions.rb
index ef0c9c5..1ffc953 100644
--- a/actionpack/lib/action_dispatch/middleware/show_exceptions.rb
+++ b/actionpack/lib/action_dispatch/middleware/show_exceptions.rb
@@ -64,7 +64,7 @@ module ActionDispatch
         log_error(exception)
 
         request = Request.new(env)
-        if @consider_all_requests_local || request.local?
+        if @consider_all_requests_local
           rescue_action_locally(request, exception)
         else
           rescue_action_in_public(exception)
diff --git a/actionpack/test/abstract_unit.rb b/actionpack/test/abstract_unit.rb
index 470b36d..8c08596 100644
--- a/actionpack/test/abstract_unit.rb
+++ b/actionpack/test/abstract_unit.rb
@@ -166,7 +166,7 @@ class ActionDispatch::IntegrationTest < ActiveSupport::TestCase
 
   def self.build_app(routes = nil)
     RoutedRackApp.new(routes || ActionDispatch::Routing::RouteSet.new) do |middleware|
-      middleware.use "ActionDispatch::ShowExceptions"
+      middleware.use "ActionDispatch::ShowExceptions", true
       middleware.use "ActionDispatch::Callbacks"
       middleware.use "ActionDispatch::ParamsParser"
       middleware.use "ActionDispatch::Cookies"
diff --git a/actionpack/test/dispatch/show_exceptions_test.rb b/actionpack/test/dispatch/show_exceptions_test.rb
index ce6c397..27a80b1 100644
--- a/actionpack/test/dispatch/show_exceptions_test.rb
+++ b/actionpack/test/dispatch/show_exceptions_test.rb
@@ -37,22 +37,22 @@ class ShowExceptionsTest < ActionDispatch::IntegrationTest
     assert_equal "", body
   end
 
-  test "rescue locally from a local request" do
+  test "rescue publically from a local request when consider_all_requests_local is false" do
     @app = ProductionApp
     ['127.0.0.1', '127.0.0.127', '::1', '0:0:0:0:0:0:0:1', '0:0:0:0:0:0:0:1%0'].each do |ip_address|
       self.remote_addr = ip_address
 
       get "/", {}, {'action_dispatch.show_exceptions' => true}
       assert_response 500
-      assert_match(/puke/, body)
+      assert_equal "500 error fixture\n", body
 
       get "/not_found", {}, {'action_dispatch.show_exceptions' => true}
       assert_response 404
-      assert_match(/#{ActionController::UnknownAction.name}/, body)
+      assert_equal "404 error fixture\n", body
 
       get "/method_not_allowed", {}, {'action_dispatch.show_exceptions' => true}
       assert_response 405
-      assert_match(/ActionController::MethodNotAllowed/, body)
+      assert_equal "", body
     end
   end
 
-- 
1.7.2.1

